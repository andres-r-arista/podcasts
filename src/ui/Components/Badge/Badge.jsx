import React from "react";

import PropTypes from "prop-types";

export const Badge = ({ label, className }) => {
  return (
    <div className={`px-1 rounded-lg self-center ${className}`} data-testid="badge">{label}</div>
  );
};

Badge.propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
};
