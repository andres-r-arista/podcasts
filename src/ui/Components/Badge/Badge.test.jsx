import { render, screen } from "@testing-library/react";

import { Badge } from "./Badge";

describe("Tests in <Badge />", () => {
  test("must match the snapshot", () => {
    const { container } = render(<Badge label={"5"} />);
    expect(container).toMatchSnapshot();
  });

  test("must print the label", () => {
    render(<Badge label={"15"} />);
    const badge = screen.getByTestId("badge");
    expect(badge).toHaveTextContent("15");
  });
});
