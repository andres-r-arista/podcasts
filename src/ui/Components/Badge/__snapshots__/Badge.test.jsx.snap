// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Badge /> must match the snapshot 1`] = `
<div>
  <div
    class="px-1 rounded-lg self-center undefined"
    data-testid="badge"
  >
    5
  </div>
</div>
`;
